README.md

```diff
@@ -1824,4 +1824,134 @@ When you are dealing with exceptions in your code, use `throw new RuntimeExcepti
 
 - It gives the programmer more control over the exception.
 - It gives the programmer the power of logging exceptions.
-- It reduces visibility of the working of your code for potential hackers.
\ No newline at end of file
+- It reduces visibility of the working of your code for potential hackers.
+
+## Unix Directories
+
+For operating systems built on Unix (e.g. Linux and Mac) the highest level of the directory, also called the root folder, is denoted by `/`.
+
+The home folder can be written as `/home/username` (e.g.`/home/jen`) or `~`.
+
+The command `pwd` (print working directory) will print the current directory to the screen.
+
+The command `echo $0` will print the name of the shell you are using.
+
+The command `echo $?` will find the value of the last executed command and return it as a number. If the value exists then a `0` will be returned. If the value does not exist a number other than `0` will be returned (which number depends on the shell you are using).
+
+The command `cat` followed by a filename will show you the contents of a file. This command only works for files (it is like `ls` for directories). Avoid `cat ls` as the result is binary.
+
+## Unix Environment Variables
+
+Environment Variables are key-value pairings where the keys can be assigned to different values over time. They are useful for software which has variables it needs to treat as constants, where the values might need to be reassigned over time.
+
+Environment Variables are also useful for storing changes that the user would like to persist in every instance of the command line.
+
+One of the most common Environment Variables is:
+
+Key: `$HOME`
+
+Value: `/home/username`
+
+You can print the `/home/username` value to the command line using `echo $HOME`.
+
+Another common Environment Variable is:
+
+Key: `$PATH`
+
+Value: A list of directories which the operating system uses to search for executable programs e.g. `/usr/local/bin`.
+
+You can print the list of directories using `echo $PATH`.
+
+You can add a directory to the list using `export PATH=/put/directory/location/here/:$PATH` or `export PATH=.:$PATH` which will add the current directory to the list.
```

Comment:
What's the difference between doing
```bash
export PATH=$PATH:$HOME/scripts
```
and
```bash
export PATH=$HOME/scripts:$PATH
```
?

Reply:
The first command adds $HOME/scripts to the end of the $PATH and the second command adds $HOME/scripts to the beginning of the $PATH.